core/group-sign_ZZZ.c.in

```diff
@@ -986,9 +959,13 @@ int GS_finishJoin(
     return GS_INVALID_USER_PRIVATE_KEY;
   }
 
+  // Temporal random number generator for fast verification, seeded with user secret.
+  csprng rng;
+  RAND_seed(&rng, len_gsk, gsk);
```

Comment:
Won't the client always get the same random sequence here, or is the client's gsk different for each day?

Reply:
This is to generate the two random factors that are required for `verifyAuxFast` (probabilistic). As long as it is unpredictable for the server, it should be fine. By seeding it with the user secret key (which is different for every day), this should hold. Also, this will only be done once per join (= per day).